packages/client/lib/rpc/modules/engine.ts

```diff
@@ -547,6 +547,17 @@ export class Engine {
   }
 
   async newPayloadV2(params: [ExecutionPayloadV2]): Promise<PayloadStatusV1> {
+    const shanghaiTimestamp = this.chain.config.chainCommon
+      .hardforks()
+      .find((hf) => hf.name === 'shanghai')?.timestamp!
```

Comment:
I guess at some point we should replace these kind of retrievals with a new method `hardforkTimestamp()` analog to `hardforkBlock()` in Common (which throws if requested for a block HF).

But I guess for now and to getting these kind of fixes in quickly it is ok.

Reply:
@g11tech added a method with that name last month in `common.ts`

https://github.com/ethereumjs/ethereumjs-monorepo/blob/7d7b0ad0915b560093a7e1ac5802432e40f436f5/packages/common/src/common.ts#L669-L676